charts/redis-ha/templates/_configs.tpl

```diff
@@ -41,20 +75,28 @@
     RO_REPLICAS="{{ .Values.ro_replicas }}"
     {{- end }}
     INDEX="${HOSTNAME##*-}"
+    SENTINEL_PORT={{ .Values.sentinel.port }}
     MASTER=''
     MASTER_GROUP="{{ template "redis-ha.masterGroupName" . }}"
     QUORUM="{{ .Values.sentinel.quorum }}"
     REDIS_CONF=/data/conf/redis.conf
     REDIS_PORT={{ .Values.redis.port }}
+    REDIS_TLS_PORT={{ .Values.redis.tlsPort }}
     SENTINEL_CONF=/data/conf/sentinel.conf
-    SENTINEL_PORT={{ .Values.sentinel.port }}
+    SENTINEL_TLS_PORT={{ .Values.sentinel.tlsPort }}
     SERVICE={{ template "redis-ha.fullname" . }}
+    TLS_REPLICATION={{ default false .Values.sentinel.tlsReplication }}
     set -eu
 
     sentinel_get_master() {
     set +e
-        redis-cli -h "${SERVICE}" -p "${SENTINEL_PORT}" sentinel get-master-addr-by-name "${MASTER_GROUP}" |\
+        if [ "$SENTINEL_PORT" -ne 0 ]; then
+            redis-cli -h "${SERVICE}" -p "${SENTINEL_PORT}" {{ if .Values.sentinel.auth }} -a "${SENTINELAUTH}"{{ end }} sentinel get-master-addr-by-name "${MASTER_GROUP}" |\
             grep -E '[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}'
+        else
+            redis-cli -h "${SERVICE}" -p "${SENTINEL_TLS_PORT}" {{ if .Values.sentinel.auth }} -a "${SENTINELAUTH}"{{ end }} {{ if ne (default "yes" .Values.sentinel.authClients) "no"}} --tls --cacert /tls-certs/{{ .Values.tls.caCertFile }} --cert /tls-certs/{{ .Values.tls.certFile }} --key /tls-certs/{{ .Values.tls.keyFile }}{{ end }} sentinel get-master-addr-by-name "${MASTER_GROUP}" |\
+            grep -E '[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}'
+        fi    
```

Comment:
Is it not better to assume the user is not using TLS, rather then the other way round? This seems dangerous.

Reply:
So I guess what you mean is that we want to check if "tls-replication" is set to yes for sentinel and then also check if sentinel port is set to 0, otherwise just connect on non-tls sentinel port?

Reply:
I'm saying we should check for TLS settings and default to non-tls, basically the opposite of this.